testutil/httptestfileservers.go

```diff
@@ -0,0 +1,21 @@
+package testutil
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func HttpTestFileServer(t *testing.T, dir string) (*httptest.Server, error) {
```

Comment:
Maybe change the signature of these 2 methods to not return an error and not have to check for it in tests?

Reply:
Hey, doing that removes the ability to call this from go-routines. I realised while writing the tests that it's always good for harness helpers to return an error so we can call it from go-routines and then assert on the error by having the error group go-routines return the error.

Reply:
WDYM? There is no error here, we always return `nil`. Why do you need the `nil` error to call this from a goroutine?